Index: src/game/Level1.cpp
===================================================================
--- src/game/Level1.cpp	(revision 5237)
+++ src/game/Level1.cpp	(working copy)
@@ -1,4 +1,4 @@
-/* 
+/*
  * Copyright (C) 2005-2008 MaNGOS <http://www.mangosproject.org/>
  *
  * This program is free software; you can redistribute it and/or modify
@@ -277,6 +277,11 @@
     Player *chr = objmgr.GetPlayer(name.c_str());
     if (chr)
     {
+        if(chr->GetSession()->GetSecurity() > m_session->GetSecurity())
+        {
+            SendSysMessage("Your GM Level Is Not High Enough To Summon This Player!");
+            return true;
+        }
         if(chr->IsBeingTeleported()==true)
         {
             PSendSysMessage(LANG_IS_TELEPORTED, chr->GetName());
@@ -364,6 +369,11 @@
     Player *chr = objmgr.GetPlayer(name.c_str());
     if (chr)
     {
+        if(chr->GetSession()->GetSecurity() > _player->GetSession()->GetSecurity())
+        {
+            SendSysMessage("Your GM Level Is Not High Enough To Teleport To That Player!");
+            return true;
+        }
         if(chr->isInFlight())
         {
             PSendSysMessage(LANG_CHAR_IN_FLIGHT, chr->GetName());
@@ -1782,7 +1792,11 @@
     Player *chr = objmgr.GetPlayer(name.c_str());
     if (chr)
     {
-
+        if(chr->GetSession()->GetSecurity() > m_session->GetSecurity())
+        {
+            SendSysMessage("Your GM Level Is Not High Enough To Teleport This Player!");
+            return true;
+        }
         if(chr->IsBeingTeleported()==true)
         {
             PSendSysMessage(LANG_IS_TELEPORTED, chr->GetName());
